vyper/compiler/output_bundle.py

```diff
@@ -0,0 +1,232 @@
+            sources[_anonymize(path)] = c
+
+        return sources
+
+    @cached_property
+    def compilation_target_path(self):
+        p = self.compiler_data.file_input.resolved_path
+        p = os.path.relpath(str(p))
+        return _anonymize(p)
+
+    @cached_property
+    def used_search_paths(self) -> list[str]:
+        # report back which search paths were "actually used" in this
+        # compilation run. this is useful mainly for aesthetic purposes,
+        # because we don't need to see `/usr/lib/python` in the search path
+        # if it is not used.
+        # that being said, we are overly conservative. that is, we might
+        # put search paths which are not actually used in the output.
+
+        search_paths = self.compiler_data.input_bundle.search_paths
+
+        tmp = {sp: 0 for sp in search_paths}
+
+        for c in self.compiler_inputs.values():
+            # recover the search path that was used for this CompilerInput.
+            # note that it is not sufficient to thread the "search path that
+            # was used" into CompilerInput because search_paths are modified
+            # during compilation (so a search path which does not exist in
+            # the original search_paths set could be used for a given file).
+            for sp in reversed(search_paths):
+                if c.resolved_path.is_relative_to(sp):
+                    # don't break here. if there are more than 1 search path
+                    # which could possibly match, we add all them to the
+                    # output.
+                    tmp[sp] += 1
+
+        sps = [sp for sp, count in tmp.items() if count > 0]
+        return [_anonymize(os.path.relpath(sp)) for sp in sps]
+
+
+class OutputBundleWriter:
+    def __init__(self, compiler_data: CompilerData):
+        self.compiler_data = compiler_data
+
+    @cached_property
+    def bundle(self):
+        return OutputBundle(self.compiler_data)
+
+    def write_sources(self, sources: dict[str, CompilerInput]):
+        raise NotImplementedError(f"write_sources: {self.__class__}")
+
+    def write_search_paths(self, search_paths: list[str]):
+        raise NotImplementedError(f"write_search_paths: {self.__class__}")
+
+    def write_settings(self, settings: Optional[Settings]):
+        raise NotImplementedError(f"write_settings: {self.__class__}")
+
+    def write_integrity(self, integrity_sum: str):
+        raise NotImplementedError(f"write_integrity: {self.__class__}")
+
+    def write_compilation_target(self, targets: list[str]):
+        raise NotImplementedError(f"write_compilation_target: {self.__class__}")
+
+    def write_compiler_version(self, version: str):
+        raise NotImplementedError(f"write_compiler_version: {self.__class__}")
+
+    def output(self):
+        raise NotImplementedError(f"output: {self.__class__}")
+
+    def write(self):
+        from vyper import __long_version__  # TODO: evil import cycle
```

Comment:
Maybe move this local import to a function in `vyper.utils`

Reply:
good idea